### ISSUE TITLE
saving state on windows-drives fails in lilv_mkdir_p

### ISSUE THREAD

**Issue by x42** [drobilla/lilv#10]
mkdir ("G:", 0755); fails with EACCESS,  likewise trying to create "G:\" also fails with EACCESS,
while creating "G:\test" works.

lilv_mkdir_p() splits the path by separator.  e.g.  "G:\test\folder\"  starts with "G:".

**Comment by drd-ableton** [comment]
Seems like the real root of this problem is attempting to create drives, which never makes sense.  Can you try https://github.com/drobilla/lilv/tree/fix-windows-mkdir ?

**Comment by drobilla** [comment]
Seems like the real root of this problem is attempting to create drives, which never makes sense. Can you try https://github.com/drobilla/lilv/tree/fix-windows-mkdir ?

**Comment by x42** [comment]
My initial solution was the same. Then I tried to come up with a method like your `is_windows_path()` and then read https://docs.microsoft.com/en-us/windows/desktop/FileIO/naming-a-file  shook my head and gave up.

I think your `is_windows_path()` is not complete, but it'll likely catch all relevant cases.  Works here with a quick test anyway.

**Comment by drobilla** [comment]
Yeah, some of the more esoteric cases won't work (relative paths that are contrained to a drive and all that craziness), but serd uses the same one (as does that other bit in lilv), so if any of these actually show up and aren't properly detected, things will probably be broken anyway.

**Comment by x42** [comment]
Are you going to apply this patch to master, and also include the fix-fake-ms branch?
